### PR TITLE
Use of Wrap as a standalone package

### PR DESCRIPTION
**Pull request by varunagrawal**
This PR establishes the independence of this project not just from GTSAM but also from needing `wrap` as a submodule.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -35,7 +43,7 @@ install(TARGETS ${PROJECT_NAME} LIBRARY DESTINATION lib ARCHIVE DESTINATION lib
 
 ###################################################################################
 # Build Pybind wrapper
-list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/wrap/cmake")
+find_package(gtwrap)
```
Comment (jlblancoc) [inline CMakeLists.txt:57]: I couldn't find where gtwrap-config.cmake or FindGtWrap.cmake are defined?

From the removed line above, I expected it to be in the wrap repo, but couldn't find it. Sorry if it's a silly question, but I never built the python wrapper for gtsam so I'm unfamiliar with it.

Please, add a comment at this point explaining what is gtwrap and where it comes from.

**Review by jlblancoc — COMMENTED** [review]
Left a comment, plus: 

> Clone this repository with all submodules by git clone --recursive https://github.com/borglab/wrap-project-python.git

This line in the README should be updated now not to use --recursive.

File: CMakeLists.txt (hunk 1 above)
Comment (varunagrawal) [reply CMakeLists.txt:57]: That's my bad. You'll have to install the PR branch version of wrap as detailed here: https://github.com/borglab/wrap/pull/11

It should have all the instructions you need to install the CMake modules.

**Comment by varunagrawal** [comment]
@ProfFan @jlblancoc could you folks test this out? Would love to get this landed.

**Review by yetongumich — COMMENTED** [review]
Everything else works except for "make python-install", which I need to run "pip3 install -e ." instead

**Comment by varunagrawal** [comment]
@yetongumich just made some updates. Can you please try with the latest?

**Pull request merged**